README.md

```diff
@@ -57,25 +57,31 @@ This guide assumes:
 
 ## Configuration
 
-Configuration is started in GitHub and completed in Mattermost. 
+GitHub plugin configuration starts by registering an OAuth app in GitHub and ends in Mattermost. 
 
 **Note:** If you're using GitHub Enterprise, replace all GitHub links below with your GitHub Enterprise URL.
 
 ### Step 1: Register an OAuth Application in GitHub
 
+You must first register the Mattermost GitHub Plugin as an authorized OAuth app regardless of whether you're setting up the GitHub plugin as a system admin or a Mattermost user.
+
 1. Go to https://github.com/settings/applications/new to register an OAuth app.
 2. Set the following values:
-   - **Application Name:** `Mattermost GitHub Plugin - <your company name>`
+   - **Application name:** `Mattermost GitHub Plugin - <your company name>`
    - **Homepage URL:** `https://github.com/mattermost/mattermost-plugin-github`
-   - **Authorization callback URL:** `https://your-mattermost-url.com/plugins/github/oauth/complete`, replacing `https://your-mattermost-url.com` with your Mattermost URL.
+   - **Authorization callback URL:** `https://your-mattermost-url.com/plugins/github/oauth/complete`, replacing `https://your-mattermost-url.com` with your Mattermost URL. This value needs to match the Mattermost server URL that you or your users users log in to. 
 3. Submit.
-4. Copy the **Client ID** and **Client Secret** in the resulting screen.
-5. Go to **System Console > Plugins > GitHub** and enter the **GitHub OAuth Client ID** and **GitHub OAuth Client Secret** you copied in a previous step.
-6. Hit **Save**.
+4. Click **Generate a new client secret** and provide your GitHub password to continue.
```

Comment:
I just tested with my account and didn't had to fill in my password. I'm concerned about confusing users. Can you just drop that part?

Reply:
I think it may be a required step the very first time you're setting things up (and perhaps your password is cached after that point?) Perhaps we could validate this step the next time someone new is attempting this?

Reply:
There might be some non deterministic behaviour from GitHub wrt the password input.